### PR TITLE
sql: use session-scoped PTS for tenant fingerprint

### PR DESCRIPTION
**Pull request by stevendanna**
Tenant-level fingerprinting may take considerable time. Further, we may be issuing a fingerprint after

Informs #110811

Release note: None

### CODE REVIEW

**Comment by stevendanna** [comment]
Note this doesn't completely fix the listed issue since we don't have revision history fingerprints in the SQL syntax yet.

File: pkg/sql/show_fingerprints.go [hunk 1]
```diff
@@ -118,6 +124,49 @@ func (n *showFingerprintsNode) startExec(params runParams) error {
 	return nil
 }
 
+// protectTenantSpanWithSession creates a protected timestamp record
+// for the given tenant ID at the read timestamp of the current
+// transaction. The PTS record will be timed to the given sessionID.
```
Comment (adityamaru) [inline pkg/sql/show_fingerprints.go:129]: s/timed/tied/g

File: pkg/sql/show_fingerprints_test.go [hunk 2]
```diff
@@ -120,3 +131,105 @@ func TestShowFingerprintsDuringSchemaChange(t *testing.T) {
...
+	})
+	require.NoError(t, err)
+
+	tenantSQL := sqlutils.MakeSQLRunner(tenantDB)
+	tenantSQL.Exec(t, "CREATE DATABASE test")
+	tenantSQL.Exec(t, "CREATE TABLE test.foo (k PRIMARY KEY) AS SELECT generate_series(1, 1000)")
+	tenantSQL.Exec(t, "ALTER TABLE test.foo CONFIGURE ZONE USING gc.ttlseconds=1")
+	tenantSQL.Exec(t, "UPDATE test.foo SET k=k+2000")
+
+	// refreshPTSReaderCache
```
Comment (adityamaru) [inline pkg/sql/show_fingerprints_test.go:189]: delete?

**Review by adityamaru — APPROVED** [review]
Really nice, the commit message seems to have gotten truncated. Also you've probably already done this but if you can run the test without the Protect changes to make sure it fails.

**Comment by stevendanna** [comment]
> Also you've probably already done this but if you can run the test without the Protect changes to make sure it fails.

Yup, did this locally.

**Comment by stevendanna** [comment]
Failure is a known issue.

bors r=adityamaru

**Comment by stevendanna** [comment]
bors cancel

**Comment by stevendanna** [comment]
bors r=adityamaru

**Pull request merged**